### PR TITLE
[#12946] Revert attributes removed in initialisation of MCS entities

### PR DESCRIPTION
**Pull request by domlimm**
Part of https://github.com/TEAMMATES/teammates/issues/12946.

**Outline of Solution**
As per title.

For reference: #12948.

### CODE REVIEW

File: src/main/java/teammates/storage/sqlentity/CourseStructure.java [hunk 1]
```diff
@@ -31,10 +31,10 @@ public class CourseStructure extends BaseEntity {
     private Course course;
 
     @Column(nullable = false)
-    private String name;
+    private String name; //NOPMD UnusedPrivateField, SingularField
```
Comment (weiquu) [inline src/main/java/teammates/storage/sqlentity/CourseStructure.java:34]: Add `@SuppressFBWarnings("URF_UNREAD_FIELD")` above `private String name;`

File: src/main/java/teammates/storage/sqlentity/CourseStructure.java [hunk 2]
```diff
@@ -31,10 +31,10 @@ public class CourseStructure extends BaseEntity {
...
 
     @OneToMany(mappedBy = "section", cascade = CascadeType.ALL)
-    private List<Section> sections = new ArrayList<>();
+    private List<Section> sections = new ArrayList<>(); //NOPMD UnusedPrivateField
```
Comment (weiquu) [inline src/main/java/teammates/storage/sqlentity/CourseStructure.java:37]: Add `@SuppressFBWarnings("URF_UNREAD_FIELD")` above `private List<Section> sections`

**Review by weiquu — CHANGES_REQUESTED** [review]
The lint issues are caused by the `spotBugs` checks, so we need to suppress the `FindBugs` warnings. See comments below, still need to add `import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;` right above `import jakarta.persistence.CascadeType;`

Otherwise looks fine

**Comment by domlimm** [comment]
@weiquu Thanks wq! Have made the changes. Will request for review once checks are completed.

**Review by weiquu — APPROVED** [review]
LGTM. Let's fix the other tests only after the current round of E2E tests migration is done

**Pull request merged**